<launch>

   <!-- Arguments that are passed from the top launchfile -->
   <arg name="vehicle_id" />
   <arg name="vehicle_ns" />

   <arg name="steering_angle_in_topic" />
   <arg name="acceleration_in_topic" />
   <arg name="objects_ground_truth_topic_with_ns" />
   <arg name="desired_motion_topic_with_ns" />

   <arg name="vehicle_parameter_file" default="$(find sim_sample_actuator_ros_tool)/launch/params/actuator_single_track_model_parameters.yaml" />

   <!-- Launch Nodes -->

   <node pkg="sim_sample_actuator_ros_tool" type="actuator_single_track_model" name="actuator_single_track_model" output="screen">
        <rosparam command="load" file="$(arg vehicle_parameter_file)"/>

        <!-- Set Parameters -->
        <param name="vehicle_id" value="$(arg vehicle_id)" />
        <param name="vehicle_ns" value="$(arg vehicle_ns)"/>

        <!-- Remap Topics -->
        <remap from="default_steering_angle_in_topic" to="/$(arg vehicle_ns)/$(arg steering_angle_in_topic)"/>
        <remap from="default_acceleration_in_topic" to="/$(arg vehicle_ns)/$(arg acceleration_in_topic)"/>
        <remap from="default_objects_ground_truth_topic_with_ns" to="$(arg objects_ground_truth_topic_with_ns)"/>
        <remap from="default_desired_motion_out_topic" to="$(arg desired_motion_topic_with_ns)"/>
   </node>

</launch>
